code/modules/power/solars/panel.dm

```diff
@@ -61,9 +60,15 @@
 
 	healthcheck()
 
+/obj/machinery/power/solar/panel/bullet_act(var/obj/item/projectile/Proj)
+	if(Proj.damage)
+		health -= Proj.damage
+		healthcheck()
+	..()
+
 /obj/machinery/power/solar/panel/proc/healthcheck()
 	if(health <= 0)
-		if(!(stat & BROKEN))
+		if(!(stat & BROKEN) && health > -maxhealth)
```

Comment:
Previously any damage that would take it to or below 0 health would set it to broken, and the next damage would destroy it and spawn some shards. Now if it takes enough to go below negative maxhealth, it just skips the broken step